src/components/Fallback/Fallback.tsx

```diff
@@ -0,0 +1,85 @@
+import { FallbackProps } from 'react-error-boundary';
+import { useNavigate } from 'react-router-dom';
+
+import ErrorPpussung from '@/assets/error_ppussung.webp';
+import {
+  StyledBoldText,
+  StyledErrorImg,
+  StyledErrorTextContainer,
+  StyledFallbackContainer,
+  StyledNavigateButton,
+  StyledSubText,
+} from '@/components/Fallback/Fallback.style';
+import { FallbackWithNavigateProps, FallbackContent } from '@/components/Fallback/Fallback.type';
+import { FALLBACK_TEXT } from '@/constants/fallback.constant';
+
+const getFallbackContent = (status: number): FallbackContent => {
+  if (status >= 500) return FALLBACK_TEXT['SERVER'];
+  if (status >= 400) return FALLBACK_TEXT['CLIENT'];
+  return { boldText: '' };
+};
+
+export const Fallback = ({ error, resetErrorBoundary }: FallbackProps) => {
+  const getBackUrl = () => {
+    if (window.location.pathname.startsWith('/drawer')) {
+      return '/drawer';
+    }
+    return '/';
+  };
+
+  const renderFallback = () => {
+    const status = error.response?.data.status;
+
+    if (status >= 500) {
+      return <FallbackWithDetail status={status} />;
+    }
+    return (
+      <FallbackWithNavigate
+        error={error}
+        resetErrorBoundary={resetErrorBoundary}
+        backUrl={getBackUrl()}
+      />
+    );
+  };
+
+  return <>{renderFallback()}</>;
```

Comment:
```suggestion
export const Fallback: React.FC<FallbackProps> = ({ error, resetErrorBoundary }) => {
  const status = error.response?.data.status;

  if (status >= 500) {
    return <FallbackWithDetail status={status} />;
  }

  return <FallbackWithNavigate error={error} resetErrorBoundary={resetErrorBoundary} backUrl={getBackUrl()} />;
};
```
이렇게 처리하면 따로 renderFallback 함수를 선언하지 않아도 될 것 같아요!

Reply:
원래 `renderFallback()` 내부가 switch문이어서 함수로 뺀 건데 조건문으로 수정하면서 함수 벗길 생각은 못했네요 🤔 확인이욥

Reply:
근데 왜 React.FC 쓰시는지 궁금해여
저는 잘 안써봐서............................................. 엠제이가 설명해주면 좋겠다. 이상입니다

Reply:
> 근데 왜 React.FC 쓰시는지 궁금해여 (수정은 React.FC 안 쓴 코드로 했어요) 저는 잘 안써봐서............................................. 엠제이가 설명해주면 좋겠다. 이상입니다

그것은 아무 컴포넌트나 열었는데 그곳에 React.FC가 있었기 때문입니다. (SearchButton)
저는 rfc로 짜잔 뿅하고 나타나는 템플릿 코드를 많이 씁니당,, 

**AI왈**: React.FC를 사용하면 children props가 내장되어 있다. propsType이나 defaultType에 대한 추론을 해준다.
**블로그왈**: children을 사용하지 않는 컴포넌트도 있어서 React.FC는 혼돈을 줄 수 이따.

Reply:
? 저희 코드에 지금 FC 쓰는 곳이 있나요? 이거 레거신데😐 이제는 작성하는 게 권장되지 않는 코드라 지우는 게 좋을 것 같네요

Reply:
> 그것은 아무 컴포넌트나 열었는데 그곳에 React.FC가 있었기 때문입니다.

이런 깊은 뜻이

Reply:
헉 한나 제가 잘못봤네여. Bar.tsx 컴포넌트에 있어여

Reply:
이 PR에 삭제 커밋 쑤셔넣는 거 어떠신가여